client/ayon_core/tools/publisher/window.py

```diff
@@ -823,15 +830,44 @@ def _set_publish_comment(self):
         self._controller.set_comment(self._comment_input.text())
 
     def _on_validate_clicked(self):
-        if self._save_changes(False):
+        if self._start_publish_preflight() and self._save_changes(False):
             self._set_publish_comment()
             self._controller.validate()
 
     def _on_publish_clicked(self):
-        if self._save_changes(False):
+        if self._start_publish_preflight() and self._save_changes(False):
             self._set_publish_comment()
             self._controller.publish()
 
+    def _start_publish_preflight(self) -> bool:
+        # Validate comment length
+        comment_def = self._controller.get_comment_def()
```

Comment:
You can cache the value in `_on_controller_reset`.

Reply:
> You can cache the value in _on_controller_reset.

I could - but then any performance hit would already happen on reset - which usually is already the slow part ;) so I wanted to avoid that. But we could 'cache' it on first query and clear cache on reset. However, the "time taken" for the response is literally very fast (because the settings used are coming from the create context cache). So it'd be a micro-optimization currently.

Can we avoid it until it's an actual issue?